source/platforms/github/_tests/_github_git.test.ts

```diff
@@ -146,6 +146,62 @@ describe("the dangerfile gitDSL", async () => {
       expect(empty).toEqual(null)
     })
 
+    it("handles showing a patch for a created file", async () => {
+      const before = ""
+
+      const after = {
+        a: "o, world",
+        b: 3,
+        c: ["one", "two", "three", "four"],
+      }
+
+      github.api.fileContents = async (path, repo, ref) => {
+        const obj = ref === masterSHA ? before : after
+        return obj === "" ? "" : JSON.stringify(obj)
+      }
+
+      const gitDSL = await github.getPlatformGitRepresentation()
+      const empty = await gitDSL.JSONPatchForFile("data/schema.json")
+      expect(empty).toEqual({
+        before: null,
+        after: after,
+        diff: [
```

Comment:
@orta Would love to get your thoughts here.

For created files, I think it makes sense to have `before` return `null` since the `before` is the absence of the file. The `diff` is _technically_ correct to, but feel unexpected and/or inconsistent with other behavior.

What do you think about `JSONPatchForFile` returning `null` for `before`, but the `diff` as the result between `{}` and `after`?

So instead of this:

```json
{
  "op": "replace",
  "path": "",
  "value": {
    "a": "o, world",
    "b": 3,
    "c": ["one", "two", "three", "four"],
  }
}
```

it becomes this:

```json
[
  {
    "op": "add",
    "path": "/a",
    "value": "o, world"
  },
  {
    "op": "add",
    "path": "/b",
    "value": 3
  },
  {
    "op": "add",
    "path": "/c",
    "value": [
      "one",
      "two",
      "three",
      "four"
    ]
  }
]
```

While not technically correct, I think it makes more sense semantically.

Reply:
I agree 👍